### PR TITLE
Bgee Adapter: gene expressed in anatomical entity

### PR DESCRIPTION
**Pull request by dawit-melka**
- implement Bgee adapter
- update biological context and ontology term representation
- user can choose writer class from command line
-  update `prolog_writer` - sanitize strings separated by comma

### CODE REVIEW

File: config/schema_config.yaml [hunk 1]
```diff
@@ -15,6 +15,27 @@ position entity:
...
+      # UMLS Semantic Group "Anatomy"
+    - UMLSSG:ANAT
+      # UMLS Semantic Type "Anatomical Structure"
+    - STY:T017
+    - FMA:62955  # anatomical entity
+    - CARO:0000000   # anatomical entity
+    - SIO:001262  # anatomical entity
+    - STY:T029  # Body Location or Region
+    - STY:T030  # Body Space or Junction
+  id_prefixes:
```
Comment (Habush) [inline config/schema_config.yaml:34]: What the purpose `id_prefixes`? Is it used in the writer class to deterimine the type of a node?

File: biocypher_metta/adapters/bgee_adapter.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        super(BgeeAdapter, self).__init__(write_properties, add_provenance)
+    
+    def get_edges(self):
+        with gzip.open(self.filepath, 'rt') as f:
+            next(f) # skip header
+            for line in f:
+                data = line.split('\t')
+                gene = data[BgeeAdapter.INDEX['gene']]
+                anatomical_entities = data[BgeeAdapter.INDEX['anatomical_entity']].split(' ∩ ')
+                score = data[BgeeAdapter.INDEX['expression_score']]
```
Comment (Habush) [inline biocypher_metta/adapters/bgee_adapter.py:30]: In addition to `anatomical_entity` and `expression_score` fields, add `FDR` field as `pvalue`.

File: biocypher_metta/adapters/bgee_adapter.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+        self.source = 'bgee'
+        self.source_url = 'https://www.bgee.org/download/gene-expression-calls?id=9606'
+        super(BgeeAdapter, self).__init__(write_properties, add_provenance)
+    
+    def get_edges(self):
+        with gzip.open(self.filepath, 'rt') as f:
+            next(f) # skip header
+            for line in f:
+                data = line.split('\t')
```
Comment (Habush) [inline biocypher_metta/adapters/bgee_adapter.py:27]: For now, we're only interested in relationships whose `Expression` column value is `present`. So filter out rows with `absent` field value in the `Expression` column.

**Review by Habush — CHANGES_REQUESTED** [review]

**Review by Habush — CHANGES_REQUESTED** [review]

File: biocypher_metta/adapters/bgee_adapter.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+                if data[BgeeAdapter.INDEX['expression']] != 'present':
+                    continue
+                gene = data[BgeeAdapter.INDEX['gene']]
+                anatomical_entities = data[BgeeAdapter.INDEX['anatomical_entity']].split(' ∩ ')
```
[the remark this answers is not in the document]
Comment (dawit-melka) [reply biocypher_metta/adapters/bgee_adapter.py:31]: There are some rows in the dataset with anatomical entity represented by`UBERON` and `CL` id separated by  this `∩` symbol (I think it is a symbol for `and` / `intersection`. I have included this row in the example sample dataset at top of the adapter.

File: config/schema_config.yaml [hunk 5]
```diff
@@ -455,6 +456,19 @@ post translational interaction:
   properties:
     score: float
 
+expressed in:
+  description: >-
+      holds between a gene and an anatomical entity in which it is expressed
```
Comment (Habush) [inline config/schema_config.yaml:461]: Can you change `anatomical entity` to `ontology term`

**Review by Habush — APPROVED** [review]

**Pull request merged**